vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,201 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewayCreate,
+		Delete: resourceVcdEdgeGatewayDelete,
+		Read:   resourceVcdEdgeGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vdc": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			/*
+					// RFC:
+				    // Waiting on a decision whether such parameters are needed
+					"delete_force": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+					"delete_recursive": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+			*/
+			"advanced_networking": &schema.Schema{
```

Comment:
"advanced_networking" -> "as_advance_gateway" to be in line with UI ?

Reply:
I have changed to "advanced" as suggested by @lvirbalas in Issue #262 comments

Reply:
We already have a vote for "advanced" in Issue #262

Reply:
@lvirbalas you  meant "advanced" or "advanced_networking" ?

Reply:
Just `advanced` (See [comment](https://github.com/terraform-providers/terraform-provider-vcd/issues/262#issuecomment-501579017))

Reply:
I am ok with only `advanced`

Reply:
changed